integration-tests/src/test/java/play/libs/ws/ahc/JsonRequestTest.java

```diff
@@ -32,4 +34,19 @@ public void setJson() throws IOException {
         assertThat(node).isEqualTo(DefaultObjectMapper.instance.readTree("{\"k1\":\"v2\"}"));
     }
 
+    @Test
+    public void test_getBodyAsJsonWithoutCharset() {
+        final Response ahcResponse = mock(Response.class);
+        final String bodyString = "{\"people\":[{\"name\":\"Jo�o\"},{\"name\":\"Fl�vio\"}]}";
```

Comment:
I just realised that won't produce the desired effect. This string is containing some 'corrupted' characters, but it's being saved in utf-8. 

It's ok to save `JsonRequestTest.java` in utf-8. That's how all files are being encoded already and we should not change it. 

Maybe instead we should move the json payload to a file on disk and save it as ISO. Then we read the file as a byte array and try to parse it.

Reply:
you're right, moved the json input to a file